internal/ingestion/knowledge_compile/writer.go

```
@@ -112,21 +191,48 @@ func (infinityWriter) DeleteMergedForDoc(ctx context.Context, tenant, kb, docID
+					continue
 				}
+				kept = append(kept, d)
+			}
+			if !changed {
+				continue
+			}
+			if len(kept) == 0 {
+				toDeleteIDs = append(toDeleteIDs, id)
+				continue
 			}
+			id, kept = id, append([]string(nil), kept...)
```

golangci-lint (2.12.2)

[error] assign: self-assignment of id

(govet)